gocourse11/filter/control_system.go

```diff
@@ -0,0 +1,42 @@
+package filter
+
+func Create(cleanLevel CleanLevel) Filter {
+	if cleanLevel == Low {
+		return NewFilter(
+			WithCleanLevel(Low),
+			WithAbsorber("sand"),
+			WithWaterImprover("t2w"),
+		)
+	}
+
+	if cleanLevel == Middle {
+		return NewFilter(
+			WithCleanLevel(Middle),
+			WithAbsorber("coal"),
+			WithWaterImprover("cn2"),
+		)
+	}
+
+	if cleanLevel == High {
+		return NewFilter(
+			WithCleanLevel(High),
+			WithAbsorber("vibranium"),
+			WithWaterImprover("yy78"),
+		)
+	}
+
+	return NewFilter(WithCleanLevel(No))
```

Comment:
Здається, `switch` тут ідеально підходить:
```suggestion
	switch cleanLevel {
	case Low:
		return NewFilter(
			WithCleanLevel(cleanLevel),
			WithAbsorber("sand"),
			WithWaterImprover("t2w"),
		)
	case Middle:
		return NewFilter(
			WithCleanLevel(cleanLevel),
			WithAbsorber("coal"),
			WithWaterImprover("cn2"),
		)
	case High:
		return NewFilter(
			WithCleanLevel(cleanLevel),
			WithAbsorber("vibranium"),
			WithWaterImprover("yy78"),
		)
	default:
		return NewFilter(WithCleanLevel(No))
	}
```